### ISSUE TITLE
Logger initialization can fail needlessly

### ISSUE THREAD

**Issue by majohn-r** [majohn-r/cmd-toolkit#31]
Logger initialization relies on finding the TMP (or TEMP) folder, which it then uses to find or create the folder where log files can be written. The current logic checks to see if TMP is defined, or then TEMP; whichever it finds first, it then checks to see whether what the variable defines is, in fact, not a plain file, and then attempts to find or create the app-specific folder for the logs. If one of those steps fails (the variable points to an existing plain file, or it the app-specific folder cannot be created), then logger initialization has failed.

But what if TMP were set to a poorly chosen value ... and TEMP was set to a well-chosen value? Well, too bad: we stopped looking when we found TMP was set to something.

Of course, reversing the search order does not help, but simply reverses the problem: TEMP seems ok, TMP really is, but we found TEMP first, so there.

Bottom line: logger initialization should be robust enough that the case of both variables being set, one to a good value, one to a poor value, should work, regardless of which variable was looked up first. If either variable has a suitable value, then just use it.
